File: docs/guides/scala/src/main/scala/globalAlgoliaUserID.scala
import algoliasearch.api.SearchClient
import algoliasearch.config.*
import algoliasearch.extension.SearchClientExtensions

def globalAlgoliaUserID(): Unit = {
  val client = SearchClient(
    appId = "ALGOLIA_APPLICATION_ID",
    apiKey = "ALGOLIA_API_KEY",
    clientOptions = ClientOptions(
      defaultHeaders = Map("X-Algolia-UserToken" -> "test-user-123")
    )
  )
  println(client)
}

File: docs/guides/javascript/src/globalAlgoliaUserID.ts
import { algoliasearch } from 'algoliasearch';

const client = algoliasearch('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY', {
  baseHeaders: {
    'X-Algolia-User-ID': 'test-user-123',
  },
});
console.log(client);
